event/nattype.go

```diff
@@ -0,0 +1,18 @@
+package event
+
+import "github.com/libp2p/go-libp2p-core/network"
+
+// EvtNATDeviceTypeChanged is an event struct to be emitted when the type of the NAT device changes for a Transport Protocol.
+//
+// Note: This event is meaningful ONLY if the AutoNAT Reachability is Private.
+// Consumers of this event should ALSO consume the `EvtLocalReachabilityChanged` event and interpret
+// this event ONLY if the Reachability on the `EvtLocalReachabilityChanged` is Private.
+type EvtNATDeviceTypeChanged struct {
```

Comment:
why `Changed`? It could be the first and only event when it is first detected, there is no change associated with it.

Reply:
Well, I mean, what if the user's WIFI changes or user connects to a new network ?

Reply:
cc @vyzo  ?

Does this sound reasonable ?

Reply:
I guess we started with `Unknown` so it does change when we detect the actual type, so *FINE*.